### DISCUSSION TITLE
mspider/best-practice/multinet-control/

### DISCUSSION THREAD

**Discussion by giscus** [DaoCloud/DaoCloud-docs#4149] [Announcements]
# mspider/best-practice/multinet-control/

DaoCloud Enterprise 5.0 下一代容器化平台集大成者，引领云原生浪潮，推动数字化转型。

https://docs.daocloud.io/mspider/best-practice/multinet-control/

**Comment by nicole-lihui** [comment]
关于多集群流量负载的用例图片中的 yaml 不正确，需要加上 outlierDetection
hello world服务的异常值检测。这是分发正常运行所必需的。特别是，它配置sidecar代理，使其知道服务的端点何时处于不健康状态。(https://istio.io/latest/docs/tasks/traffic-management/locality-load-balancing/distribute/)

```bash

$ kubectl --context="${CTX_PRIMARY}" apply -n sample -f - <<EOF
apiVersion: networking.istio.io/v1beta1
kind: DestinationRule
metadata:
  name: helloworld
spec:
  host: helloworld.sample.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      localityLbSetting:
        enabled: true
        distribute:
        - from: region1/zone1/*
          to:
            "region1/zone1/*": 70
            "region1/zone2/*": 20
            "region3/zone4/*": 10
    outlierDetection:
      consecutive5xxErrors: 100
      interval: 1s
      baseEjectionTime: 1m
EOF
```

**Reply by nicole-lihui** [reply]
outlierDetection 配置在多集群负载策略上，是有助于网格排除故障节点，是建议配置条件。
如果开启集群负载转移策略 outlierDetection 是必要条件。
